src/App.tsx

```diff
@@ -1,16 +1,41 @@
-import { Route, Routes } from "react-router-dom";
-import Body from "./components/Body";
+import { Route, Routes, useNavigate } from "react-router-dom";
+import { onAuthStateChanged } from "firebase/auth";
+import { useDispatch } from "react-redux";
+import { useEffect } from "react";
 import "./index.css";
+import Body from "./components/Body";
 import Login from "./components/Login";
 import ForgotPassword from "./components/ForgotPassword";
+import { auth } from "./utils/firebase";
+import { addUser, removeUser } from "./utils/userSlice";
+import Browse from "./components/Browse";
 
 function App() {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  useEffect(() => {
+    onAuthStateChanged(auth, (user) => {
+      if (user) {
+        //user signed in
+        const { uid, email, displayName } = user;
+        console.log(`user signed in : uid = ${uid}, email= ${email}, displayName= ${displayName}`);
+        dispatch(addUser({uid: uid, email: email, displayName: displayName}));
```

Comment:
since the key and value pair name is the same, the parameter can take just the key to understand. Makes it more readable

Reply:
updated